### PR TITLE
feat: pokemon details

### PR DESCRIPTION
**Pull request by Miqqy0069**
- added pokemon details connector, vm and page
- created reusable pokemon details cards
- added spacing constants
- ran analyze with 0 issues
- dart format with 0 changes

(analyze) 
image

(dart format) 
image

(working app)

https://user-images.githubusercontent.com/75653480/228209964-034ad44a-c705-4508-bdfd-38c3e962a318.mp4

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -13,7 +15,15 @@ void main() {
...
-      child: const PokemonOverviewConnector(),
+      child: MaterialApp(
+        routes: routes,
+        theme: ThemeData(primarySwatch: Colors.red),
+      ),
     ),
   );
 }
+
+final routes = {
```
Comment (shawn-ffufm) [inline lib/main.dart]: kindly move this in a separate dart file 🙂

File: lib/main.dart [hunk 2]
```diff
@@ -13,7 +15,15 @@ void main() {
...
+  '/': (BuildContext context) => const PokemonOverviewConnector(),
```
Comment (shawn-ffufm) [inline lib/main.dart]: rename context to _ since it's unused 🙂

File: lib/main.dart [hunk 3]
```diff
@@ -13,7 +15,15 @@ void main() {
...
+  PokemonDetailsConnector.routeName: (BuildContext context) => const PokemonDetailsConnector()
```
Comment (shawn-ffufm) [inline lib/main.dart]: also here

File: lib/state/action/pokemon_actions.dart [hunk 4]
```diff
@@ -32,3 +34,9 @@ class GetPokemonDetailsAction extends ReduxAction<AppState> {
     return state.copyWith(pokemonDetails: pokemonDetails);
   }
 }
+
+/// Dispose PokemonDetailsAction on store
```
Comment (shawn-ffufm) [inline lib/state/action/pokemon_actions.dart]: kindly improve this documentation 🙂

File: lib/utilities/spacing.dart [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import 'package:flutter/material.dart';
+
+class HorizontalSpace extends StatelessWidget {
+  final double width;
```
Comment (shawn-ffufm) [inline lib/utilities/spacing.dart]: move this below the constructor 🙂

File: lib/utilities/spacing.dart [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import 'package:flutter/material.dart';
+
+class HorizontalSpace extends StatelessWidget {
+  final double width;
+  const HorizontalSpace({
+    super.key,
+    required this.width,
```
Comment (shawn-ffufm) [inline lib/utilities/spacing.dart:19]: please make required properties always on top

File: lib/utilities/spacing.dart [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  const HorizontalSpace({
+    super.key,
+    required this.width,
+  });
+  @override
+  Widget build(BuildContext context) => SizedBox(width: width);
+}
+
+class VerticalSpace extends StatelessWidget {
+  final double height;
```
Comment (shawn-ffufm) [inline lib/utilities/spacing.dart]: also here

File: lib/utilities/spacing.dart [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import 'package:flutter/material.dart';
+
+class HorizontalSpace extends StatelessWidget {
+  final double width;
+  const HorizontalSpace({
+    super.key,
+    required this.width,
+  });
+  @override
```
Comment (shawn-ffufm) [inline lib/utilities/spacing.dart]: kindly add new line on top of this 🙂

File: lib/utilities/spacing.dart [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  });
+  @override
+  Widget build(BuildContext context) => SizedBox(width: width);
+}
+
+class VerticalSpace extends StatelessWidget {
+  final double height;
+  const VerticalSpace({
+    super.key,
+    required this.height,
```
Comment (shawn-ffufm) [inline lib/utilities/spacing.dart]: same here

File: lib/utilities/spacing.dart [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+  });
+  @override
```
Comment (shawn-ffufm) [inline lib/utilities/spacing.dart:25]: also here

**Review by shawn-ffufm — CHANGES_REQUESTED** [review]
see pr feedback 🙂

File: lib/utilities/spacing.dart (hunk 10 above)
Comment (Miqqy0069) [reply lib/utilities/spacing.dart:25]: resolved

File: lib/utilities/spacing.dart (hunk 9 above)
Comment (Miqqy0069) [reply lib/utilities/spacing.dart]: resolved

File: lib/utilities/spacing.dart (hunk 8 above)
Comment (Miqqy0069) [reply lib/utilities/spacing.dart]: resolved

File: lib/utilities/spacing.dart (hunk 7 above)
Comment (Miqqy0069) [reply lib/utilities/spacing.dart]: resolved

File: lib/utilities/spacing.dart (hunk 6 above)
Comment (Miqqy0069) [reply lib/utilities/spacing.dart:19]: resolved

File: lib/utilities/spacing.dart (hunk 5 above)
Comment (Miqqy0069) [reply lib/utilities/spacing.dart]: resolved

File: lib/state/action/pokemon_actions.dart (hunk 4 above)
Comment (Miqqy0069) [reply lib/state/action/pokemon_actions.dart]: resolved

File: lib/main.dart (hunk 3 above)
Comment (Miqqy0069) [reply lib/main.dart]: resolved

File: lib/main.dart (hunk 2 above)
Comment (Miqqy0069) [reply lib/main.dart]: resolved

File: lib/main.dart (hunk 1 above)
Comment (Miqqy0069) [reply lib/main.dart]: resolved

**Review by shawn-ffufm — APPROVED** [review]
lgtm

**Pull request merged**